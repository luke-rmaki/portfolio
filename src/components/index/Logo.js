import React, { useEffect, useState, useRef } from 'react';
import styled from 'styled-components';

const LogoImg = styled.img`
  width: 100%;
  position: absolute;
  left: 50%;
  transform: translateX(-50%);
`;

const Logo = () => {
  // Frame number to load in img source
  const [frame, setFrame] = useState(0);

  // Use useRef for mutable variables that we want to persist
  // without triggering a re-render on their change
  const requestRef = useRef();
  const currentFrame = useRef(0);

  const animate = () => {
    // Use callback for setState to ensure current state is accurate
    setFrame(() => {
      // If currentFrame is less than or equal to 48, set new state to equal currentFrame. Else set it to 48 (last frame)
      const newState =
        Math.trunc(currentFrame.current) <= 48
          ? Math.trunc(currentFrame.current)
          : 48;
      return newState;
    });
    requestRef.current = requestAnimationFrame(animate);
    currentFrame.current = currentFrame.current + 1;
  };

  useEffect(() => {
    requestRef.current = requestAnimationFrame(animate);
    return () => cancelAnimationFrame(requestRef.current);
  }, []);

  return (
    <div>
      <LogoImg
        src={`./vector_frames/frame${frame < 10 ? '0' : ''}${frame}.svg`}
      />
    </div>
  );
};

export default Logo;
