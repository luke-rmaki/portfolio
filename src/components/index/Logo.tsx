import * as React from 'react';
import styled from 'styled-components';
import { useSpring, useTrail, animated } from 'react-spring';

import colors from '../../utils/colors';

const Name: any = styled(animated.h2)`
  font-family: 'Montserrat', sans-serif;
  color: ${colors.white};
  font-size: 30px;
  white-space: nowrap;

  @media screen and (max-width: 500px) {
    font-size: 20px;
  }
  @media screen and (max-width: 300px) {
    font-size: 15px;
  }
`;

const LogoWrapper: any = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  padding-top: 100px;

  svg {
    width: 20%;
    max-height: 200px;
  }
`;

const Logo: React.FunctionComponent = () => {
  const nameAnimation: React.CSSProperties = useSpring({
    transform: 'translateY(0%)',
    from: { transform: 'translateY(-1000%)' },
    config: {
      duration: 2000,
      easing: t =>
        t < 0.5 ? 16 * t * t * t * t * t : 1 + 16 * --t * t * t * t * t,
    },
  });
  const leftAnimation: React.CSSProperties = useSpring({
    transform: 'translateX(10%)',
    from: { transform: 'translateX(100%)' },
    config: {
      duration: 1000,
    },
  });
  const rightAnimation: React.CSSProperties = useSpring({
    transform: 'translateX(-10%)',
    from: { transform: 'translateX(-100%)' },
    config: {
      duration: 1000,
    },
  });
  return (
    <LogoWrapper>
      <animated.svg
        width="100%"
        height="100%"
        viewBox="0 0 163 321"
        style={leftAnimation}
      >
        <path
          d="M160.5,2l-158.5,158.5l158.5,158.5"
          style={{ fill: 'none', stroke: '#6845e6', strokeWidth: '4px' }}
        />
      </animated.svg>
      <Name style={nameAnimation}>Luke Ruokaism&#228;ki</Name>
      <animated.svg
        width="100%"
        height="100%"
        viewBox="0 0 163 321"
        style={rightAnimation}
      >
        <path
          d="M2,319l158.5,-158.5l-158.5,-158.5"
          style={{ fill: 'none', stroke: '#6845e6', strokeWidth: '4px' }}
        />
      </animated.svg>
    </LogoWrapper>
  );
};

export default Logo;
