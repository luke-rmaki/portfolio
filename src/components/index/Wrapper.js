import styled from 'styled-components';
import Wrapper from '../styled_components/layout/Wrapper';

const IndexWrapper = styled(Wrapper)`
  position: absolute;
  bottom: ${props => props.position.bottom}px;
  left: ${props => props.position.left}px;
`;

export default IndexWrapper;
