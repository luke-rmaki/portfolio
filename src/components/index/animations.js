import React from 'react';
import { useSpring, animated } from 'react-spring';

export const AnimateLeft = props => {
  const animation = useSpring({
    from: { transform: 'translateX(200%)', opacity: 0 },
    to: { transform: 'translatex(0%)', opacity: 1 },
  });
  return <animated.div style={animation}>{props.children}</animated.div>;
};

export const AnimateCenter = props => {
  const animation = useSpring({
    from: { transform: 'translateY(200%)', opacity: 0 },
    to: { transform: 'translateY(0%)', opacity: 1 },
  });
  return <animated.div style={animation}>{props.children}</animated.div>;
};

export const AnimateRight = props => {
  const animation = useSpring({
    from: { transform: 'translateX(-200%)', opacity: 0 },
    to: { transform: 'translatex(0%)', opacity: 1 },
  });
  return <animated.div style={animation}>{props.children}</animated.div>;
};
