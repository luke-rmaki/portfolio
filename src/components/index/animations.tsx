import * as React from 'react';
import { useSpring, animated } from 'react-spring';

interface Props {
  children: React.ReactNode;
}

export const AnimateLeft: React.FunctionComponent = (props: Props) => {
  const animation: React.CSSProperties = useSpring({
    from: { transform: 'translateX(200%)', opacity: 0 },
    to: { transform: 'translatex(0%)', opacity: 1 },
  });
  return <animated.div style={animation}>{props.children}</animated.div>;
};

export const AnimateCenter: React.FunctionComponent = (props: Props) => {
  const animation: React.CSSProperties = useSpring({
    from: { transform: 'translateY(-400%)', opacity: 0 },
    to: { transform: 'translateY(0%)', opacity: 1 },
  });
  return <animated.div style={animation}>{props.children}</animated.div>;
};

export const AnimateRight: React.FunctionComponent = (props: Props) => {
  const animation: React.CSSProperties = useSpring({
    from: { transform: 'translateX(-200%)', opacity: 0 },
    to: { transform: 'translatex(0%)', opacity: 1 },
  });
  return <animated.div style={animation}>{props.children}</animated.div>;
};
