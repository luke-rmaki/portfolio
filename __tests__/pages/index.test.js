import React from 'react';
import renderer from 'react-test-renderer';

import index from '../../src/pages/index';

describe('index', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<index />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
