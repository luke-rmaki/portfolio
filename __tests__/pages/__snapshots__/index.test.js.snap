// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index renders correctly 1`] = `<index />`;
